cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -99,10 +124,20 @@ func Run(gadgetRoot, device string, options *Options) error {
 		}
 	}
 
+	// store the encryption key as the last part of the process to reduce the
+	// possiblity of exiting with an error after the TPM provisioning
+	if options.Encrypt {
+		if err := key.Store(options.KeyFile); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
 func ensureLayoutCompatibility(gadgetLayout *gadget.LaidOutVolume, diskLayout *partition.DeviceLayout) error {
+	// XXX: the compatibility tests don't recognize encrypted partitions, so we
```

Comment:
@bboozzoo @cmatsuoka what concretely do we need to do here? (I suppose this should be a TODO:UC20: ? )

Reply:
We could either declare that this case is not supported (we don't have this scenario) or we can, for the sake of tool completeness, add some mechanism to verify it (e.g. via `luksUUID` and `/dev/disk/by-d` if it's open). Given our needs and time constraints, perhaps we could leave it for a later, post-UC20 implementation?

Reply:
what does "we don't have this case" mean here?  that gadgets don't put assets in the partitions that get encrypted?

Reply:
The image file doesn't have any encrypted partitions before we create ubuntu-boot and ubuntu-data, so we don't verify consistency of encrypted partitions in our practical case (we only verify the consistency of regular, non-encrypted partitions).

Reply:
what happens if we have a reboot after we have done some of the work (and created the encrypted part) but not all of the work? cc @bboozzoo

Reply:
In this case, the partition and the fs inside are completely opaque to us. The only way we could identify it by partition label which is set at the GPT level and comes from the `name: <..>` structure attribute. I'll sync with @cmatsuoka about the handling in such scenario

Reply:
If we reboot before storing the key, we need to recreate the encrypted partitions. Rebooting right after storing the key should be the normal course of action. The reboot could also happen in mid-mkfs or mid-deploy, in which case these steps would need to be re-executed. Removing all partitions created at install time and starting over seems to be the safest to me, even in situations where no encryption is involved. (Partitions created at install time would have to be marked as so.)

Reply:
Remove the consistency note and add TODO:UC20: notes to mark newly created partitions and remove them if they exist at the start of the installation procedure. This will be implemented in a forthcoming PR.